src/cart/utils/crt_utils.c

```diff
@@ -409,6 +410,46 @@ crtu_dc_mgmt_net_cfg_rank_add(const char *name, crt_group_t *group,
 	return rc;
 }
 
+static inline void
+crtu_dc_mgmt_net_print_env(void)
+{
+	static const char *var_names[] = {"OFI_INTERFACE", "OFI_DOMAIN", "CRT_PHY_ADDR_STR",
+					  "CRT_CTX_SHARE_ADDR", "CRT_TIMEOUT"};
+	int                idx;
+	char              *env;
+	char              *msg;
+	int                rc;
+
+	rc = d_agetenv_str(&env, var_names[0]);
+	D_ASSERTF(env != NULL, "Can not retrieve environment varirable %s: " DF_RC "\n",
+		  var_names[0], DP_RC(rc));
+	D_ASPRINTF(msg, "CaRT env setup with:\n\t%s=%s", var_names[0], env);
+	d_freeenv_str(&env);
+	if (msg == NULL) {
+		D_INFO("Error allocating CaRT env setup message");
+		return;
+	}
+
+	for (idx = 1; idx < sizeof(var_names) / sizeof(char *); ++idx) {
+		char *tmp = msg;
+
+		rc = d_agetenv_str(&env, var_names[idx]);
+		D_ASSERTF(env != NULL, "Can not retrieve environment varirable %s: " DF_RC "\n",
+			  var_names[idx], DP_RC(rc));
+
+		D_ASPRINTF(msg, "%s, %s=%s", tmp, var_names[idx], env);
+		d_freeenv_str(&env);
+		D_FREE(tmp);
+		if (msg == NULL) {
+			D_INFO("Error allocating CaRT env setup message");
+			return;
+		}
+	}
+
+	D_INFO("%s", msg);
```

Comment:
Doing logging this way is both more complex but also doesn't work well with logs where multiple clients write to the same log so simply calling D_INFO per entry would be preferable.

We already print all env vars however don't we, so why print a subset of them again?

Reply:
These environment variables were printed at the end of the `crtu_dc_mgmt_net_cfg_setenv()`.
It was using `getenv()` and thus do not have to take care of memory free.
Thus, I introduced the `crtu_dc_mgmt_net_print_env()` function to be as less as possible intrusive in `crtu_dc_mgmt_net_cfg_setenv()` function.
Another solution less complex but a little bit more intrusive, could be to move the deallocation of the environment variables in the cleanup section.

Reply:
- [ ] Simplify debug information printing

Reply:
- [ ] Simplify debug information printing

Reply:
> * [x]  Simplify debug information printing

Fixed with commit a63fb0d2a3032fa22eca37f6052ad69fd3fe2782